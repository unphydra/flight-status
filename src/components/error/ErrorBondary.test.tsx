import { render, screen } from "@testing-library/react";
import ErrorBoundary from "./ErrorBoundary";

function ErrorChildComponent(): JSX.Element {
  throw new Error("Oops! Something went wrong.");
}

function FallbackComponent(): JSX.Element {
  return <div>Something went wrong!</div>;
}

describe("ErrorBoundary", () => {
  it("Should renders without crashing", () => {
    const { container } = render(
      <ErrorBoundary fallback={<FallbackComponent />}>
        <div>Normal Child Component</div>
      </ErrorBoundary>
    );
    expect(container).toMatchSnapshot();
  });

  it("should render the fallback UI when an error occurs", () => {
    render(
      <ErrorBoundary fallback={<FallbackComponent />}>
        <ErrorChildComponent />
      </ErrorBoundary>
    );
    const fallbackText = screen.getByText(/Something went wrong!/i);
    expect(fallbackText).toBeInTheDocument();
  });

  it("should render the children when no error occurs", () => {
    render(
      <ErrorBoundary fallback={<FallbackComponent />}>
        <div>Normal Child Component</div>
      </ErrorBoundary>
    );
    const normalChildComponent = screen.getByText(
      /Normal Child Component/i
    );
    expect(normalChildComponent).toBeInTheDocument();
  });
});
