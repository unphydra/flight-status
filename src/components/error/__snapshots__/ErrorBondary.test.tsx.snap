// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorBoundary > Should renders without crashing 1`] = `
<div>
  <div>
    Normal Child Component
  </div>
</div>
`;
